File: design-patterns/ts/interpreter/logical-operation/terminal-operation.ts
// terminal-operation.ts

import Operation from "./operation";

class TerminalOperation implements Operation {
    private data: string;

    constructor(data: string) {
        this.data = data;
    }

    execute(opContext: string): boolean {
        return opContext.includes(this.data);
    }
}

export default TerminalOperation;

File: design-patterns/go/interpreter/logical_operation/terminal_operation.go
// terminal_operation.go

package main

import "strings"

type TerminalOperation struct {
	data string
}

func NewTerminalOperation(data string) (terminalOperation *TerminalOperation) {
	terminalOperation = &TerminalOperation{}
	terminalOperation.data = data
	return
}

func (terminalOperation *TerminalOperation) Execute(opContext string) bool {
	return strings.Contains(opContext, terminalOperation.data)
}
